packages/vsce/src/resources/SessionHandler.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { IProfile, IProfileLoaded } from "@zowe/imperative";
+import { imperative } from "@zowe/zowe-explorer-api";
+import { CICSSession } from "./CICSSession";
+import { ISessionHandler } from "./ISessionHandler";
+
+export class SessionHandler implements ISessionHandler {
+  public sessions: Map<String, CICSSession>;
+  session: CICSSession | undefined;
+  profile: imperative.IProfile;
+  private static instance: SessionHandler;
+
+  private constructor() {}
+  // Creating a singleton instance of SessionHandler
+  public static getInstance(): SessionHandler {
+    if (!SessionHandler.instance) {
+      SessionHandler.instance = new SessionHandler();
+      SessionHandler.instance.sessions = new Map<String, CICSSession>();
+    }
+    return SessionHandler.instance;
+  }
+
+  createSession(profile: IProfile, profileName: string): void {
```

Comment:
Is the idea that this is only called from inside this class, from `getSession` if it doesn't exist?? If so, private?